File: backend/apps/automations/api/serializers/automation.py
from rest_framework import serializers
from apps.automations.models import Automation
from apps.departments.models import Department
from apps.users.models import User
from apps.positions.api.serializers import PositionSerializer


class UserSerializerAutomations(serializers.ModelSerializer):
    class Meta:
        model = User
        fields = [
            "first_name",
            "last_name",

        ]


class DepartmentSerializerAutomations(serializers.ModelSerializer):
    class Meta:
        model = Department
        fields = [
            "name",
        ]


class AutomationSerializer(serializers.ModelSerializer):
    updated_by_data = UserSerializerAutomations(
        source='updated_by', read_only=True)
    department_data = DepartmentSerializerAutomations(
        source='department', read_only=True)
    position_data = PositionSerializer(source='position', read_only=True)

    class Meta:
        model = Automation
        fields = [
            'id',
            'name',
            'description',
            'is_active',
            'created_at',
            'updated_at',
            'updated_by',
            'updated_by_data',
            'department',
            'department_data',
            'manual_time',
            'in_manutention',
            "position",
            "position_data",
            "auth_certificate",
            "url_certificate"
        ]
        read_only_fields = [
            'created_at',
            'updated_at',
            'updated_by',
            'updated_by_data',
            'department_data',
            "position_data",
        ]

File: frontend/src/types/automation.ts
import { Position } from "./position";

export interface Automation {
  id: number;
  name: string;
  description: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  manual_time: number;
  in_manutention: boolean;
  updated_by: number | null;
  updated_by_data: { first_name: string; last_name: string } | null;
  department: number | null;
  department_data: { name: string } | null;
  position: number;
  position_data: Position | null;
  auth_certificate: boolean;
  url_certificate: string;
}

export interface AutomationCreate {
  name: string;
  description?: string;
  is_active?: boolean;
  manual_time?: number;
  in_manutention?: boolean;
  department?: number;
  position?: number;
  auth_certificate?: boolean;
  url_certificate?: string;
}

export interface AutomationUpdate extends Partial<AutomationCreate> { }

export interface AutomationFilters {
  name?: string;
  is_active?: boolean;
}

export interface AutomationsStatus {
  id: number;
  name: string;
  last_status: 'iniciado' | 'concluido' | 'erro' | 'alerta' | null;
  last_date_start: string | null;
  last_date_end: string | null;
  last_execution_id: number | null;
}

export interface AutomationDetails {
  name: string;
  description: string | null;
  is_active: boolean;
}